libs-scala/gatling-utils/src/test/scala/com/daml/gatling/stats/SimulationLogSpec.scala

```diff
@@ -0,0 +1,463 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.gatling.stats
+
+import java.io.File
+
+import org.scalactic.TypeCheckedTripleEquals
+import scalaz.Scalaz._
+
+import scala.util.Random
+import com.daml.gatling.stats.util.ReadFileSyntax._
+import org.scalatest.{FlatSpec, Matchers}
+import com.daml.bazeltools.BazelRunfiles.requiredResource
+
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+class SimulationLogSpec extends FlatSpec with Matchers with TypeCheckedTripleEquals {
+  import SimulationLog._
+
+  behavior of "SimulationLog"
+
+  private val simulationLog = "libs-scala/gatling-utils/src/test/resources/simulation-log"
+
+  private def resultFor(fileName: String) =
+    SimulationLog.fromFile(requiredResource(s"$simulationLog/$fileName.txt"))
+
+  it should "fail if file does not exist" in {
+    SimulationLog.fromFile(new File("DOES-NOT-EXIST-OgUzdJsvKHc9TtfNiLXA")) shouldBe 'left
+  }
+
+  it should "fail if no RUN entry" in {
+    resultFor("no-run") shouldBe 'left
+  }
+
+  it should "fail if no USER entry" in {
+    resultFor("no-user") shouldBe 'left
+  }
+
+  it should "fail if multiple RUN entries" in {
+    resultFor("multiple-run") shouldBe 'left
+  }
+
+  it should "return correct result for minimal log" in {
+    val request = RequestStats(
+      userId = 1,
+      requestLabel = "fix-trade-sync",
```

Comment:
Probably best to do a search and replace for "fix trade".

Reply:
it is just a label in the `simulation.log`. However if you really think we should remove all references to fix-trade... It can be done. I just don't really see any value of doing this. Can be `foo` or `bar` right :)